src/Microsoft.Data/BaseColumn.cs

```diff
@@ -64,24 +67,26 @@ protected set
         /// <returns></returns>
         public virtual BaseColumn Clone(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
 
+        public virtual BaseColumn CloneAndAppendNulls(BaseColumn mapIndices = null, bool invertMapIndices = false) => throw new NotImplementedException();
```

Comment:
Can `appendNulls` just be an option to the existing `Clone` method?

Reply:
It could. I'll put it in https://github.com/dotnet/corefxlab/pull/2687